src/terminal/parser/stateMachine.cpp

```diff
@@ -2161,11 +2159,12 @@ template<typename TLambda>
 bool StateMachine::_SafeExecute(TLambda&& lambda)
 try
 {
-    return lambda();
-}
-catch (const ShutdownException&)
-{
-    throw;
+    const auto ok = lambda();
+    if (!ok)
+    {
+        FlushToTerminal();
+    }

```

Comment:
In the input state machine, `ActionDcsDispatch` is always going to return a nullptr, so the lambda call here will always be false and flush, but we don't actually want that. That's what is making the DCS reports repeat the introducer sequence from within WSL. If I make the `ActionDcsDispatch` call bypass the `_SafeExecute` then those reports seem to work again. They still don't work in a cmd shell, but that's a separate issue I think.